util/dvsim/Deploy.py

```diff
@@ -492,6 +493,8 @@ def _define_attrs(self):
             "run_cmd": False,
             "run_opts": False,
             "post_run_cmds": False,
+            "build_seed": True,  # Already set in the constructor.
```

Comment:
This does not seem right to me. `build_seed` is a SV compile-time property, its not a simulation runtime property.

Reply:
right, but the issue is that Bazel builds run during the dvsim "run phase", and in order for us to use Bazel to build specific OTP images (which we need in order to run ROM E2E tests in DV without having to specify the same OTP configuration in multiple places which can easily get out of sync) then we need to pass the build_seed to the run phase too so Bazel knows what build seed was used during SV compile time so it can use the same seed when generating the OTP image (we need to know two different seeds to build the OTP image, one is the build_seed). Does that make sense?

Reply:
If you set it to False, it seems like it will not attempt to fetch the seed value from the hjson data?

Reply:
This is set to true, since the object constructor sets their values. If these are set to false, then the `_extract_attrs` will expect them to be in the dict provided. But the `build_seed` comes from the `sim_cfg` [object](https://cs.opensource.google/opentitan/opentitan/+/master:util/dvsim/Deploy.py;drc=8e586da52d5d7bdeb3811d894357b48ad48e21e2;l=472), not the test [object](https://cs.opensource.google/opentitan/opentitan/+/master:util/dvsim/Deploy.py;drc=8e586da52d5d7bdeb3811d894357b48ad48e21e2;l=470), which is what is passed to [`_extract_attrs`](https://cs.opensource.google/opentitan/opentitan/+/master:util/dvsim/Deploy.py;drc=8e586da52d5d7bdeb3811d894357b48ad48e21e2;l=514), so we would get a key not found error.